### PR TITLE
Update Colorado TABOR parameters 2023

### LINKED ISSUE #3107: Update Colorado TABOR parameters 2023
https://github.com/PolicyEngine/policyengine-us/tree/master/policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount

https://tax.colorado.gov/cash-back

bill - https://leg.colorado.gov/sites/default/files/documents/2023A/bills/fn/2023a_hb1311_00.pdf

explainer - https://www.bellpolicy.org/2023/05/22/tabor-in-2023/

### PR DESCRIPTION
**Pull request by Eleanor-Zhu**
Fixes #3107

### Summary
📝💰📈

Update Colorado sales tax refund parameters for 2023. Modify `amount` and `threshold` fields and add `source` field in `base.yaml` to reflect new legislation.

> _The state of Colorado has passed_
> _A new law that affects sales tax_
> _It changes the `amount`_
> _And the `threshold` count_
> _And adds a new `source` to the facts_

### Walkthrough
*  Add and update sales tax refund parameters for Colorado income tax ([link](https://github.com/PolicyEngine/policyengine-us/pull/3126/files?diff=unified&w=0#diff-ab72d0ca631a6efd3a95d4986340c9fdfa54b5dfafe668aeeecc18bdbed603bbR10-R11), [link](https://github.com/PolicyEngine/policyengine-us/pull/3126/files?diff=unified&w=0#diff-ab72d0ca631a6efd3a95d4986340c9fdfa54b5dfafe668aeeecc18bdbed603bbL20-R62))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml:1]: need changelog_entry.yaml

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Update Colorado TABOR parameters 2023
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Update Colorado TABOR parameters for 2023.
```

File: changelog_entry.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:3]:
```suggestion
    fixed:
```

File: changelog_entry.yaml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]:
```suggestion
- bump: patch
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml:1]: could you add one unit test for 2023 values?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml [hunk 4]
```diff
@@ -17,33 +19,44 @@ brackets:
     amount:
       2021-01-01: 37
       2022-01-01: 153
+      2023-01-01: 454
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml:22]: please use the amounts with the new Bill: it was signed into law and the amounts increased

image

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/base.yaml]: the second column is the amount that is applicable not the first

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund.py [hunk 5]
```diff
@@ -11,12 +11,14 @@ class co_sales_tax_refund(Variable):
...
-        agi = tax_unit("co_modified_agi", period)
         multiplier = p.multiplier[filing_status]
-        amount = p.base.calc(agi)
-        return multiplier * amount
+        if not p.flat_amount_enabled:
+            filing_status = tax_unit("filing_status", period)
+            agi = tax_unit("co_modified_agi", period)
+            multiplier = p.multiplier[filing_status]
+            amount = p.scale.calc(agi)
+            return multiplier * amount
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund.py:22]:
```suggestion
```

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund.py [hunk 6]
```diff
@@ -11,12 +11,14 @@ class co_sales_tax_refund(Variable):
...
             period
         ).gov.states.co.tax.income.credits.sales_tax_refund.amount
-        agi = tax_unit("co_modified_agi", period)
         multiplier = p.multiplier[filing_status]
-        amount = p.base.calc(agi)
-        return multiplier * amount
+        if not p.flat_amount_enabled:
+            filing_status = tax_unit("filing_status", period)
+            agi = tax_unit("co_modified_agi", period)
+            multiplier = p.multiplier[filing_status]
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund.py]:
```suggestion
```

File: policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/scale.yaml [hunk 7]
```diff
@@ -7,6 +7,9 @@ metadata:
   reference:
     - title: Colo. Code Regs. 39-22-2003 State Sales Tax Refund (1), (b)
       href: https://casetext.com/regulation/colorado-administrative-code/department-200-department-of-revenue/division-201-taxation-division/rule-1-ccr-201-2-income-tax/rule-39-22-2003-effective-until-8302023-state-sales-tax-refund
+    # In 2023 Colorado switched to a flat amount as opposed to a scale based on income.
+    - title: 2023 Colorado Individual Income Tax Filing Guide
+      href: https://leg.colorado.gov/sites/default/files/documents/2023A/bills/fn/2023a_hb1311_00.pdf#page=3
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/co/tax/income/credits/sales_tax_refund/amount/scale.yaml:12]: Please add a note or source showing that this bill was passed and signed by the governor.

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund.py [hunk 8]
```diff
@@ -11,12 +11,16 @@ class co_sales_tax_refund(Variable):
...
+        filing_status = tax_unit("filing_status", period)
         multiplier = p.multiplier[filing_status]
-        amount = p.base.calc(agi)
+        if p.flat_amount_enabled:
+            amount = p.amount
+        else:
+            agi = tax_unit("co_modified_agi", period)
+            multiplier = p.multiplier[filing_status]
+            amount = p.scale.calc(agi)
+            return multiplier * amount
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/co/tax/income/credits/sales_tax_refund/co_sales_tax_refund.py:25]:
```suggestion
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
Last thing then good to merge

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**